### ISSUE TITLE
Breakpoints don't work on a mounted dev drive

### ISSUE THREAD

**Issue by mvarendorff2** [flutter/flutter-intellij#7336]
## Steps to Reproduce

1. On Windows 11, create a Dev Drive (let's call it D: here)
2. Mount D: to a subfolder of C: (for example C:\Projects)
3. Create a project in D:\my-great-flutter-project
4. Open it in Android Studio using the path C:\Projects\my-great-flutter-project
5. Try to debug the app with breakpoints -> doesn't work and breakpoints are shown as 🚫 :( 
6. Open it in Android Studio using the path D:\my-great-flutter-project
7. Try to debug the app with breakpoints -> works :)

## Version info

<details>

<summary> Flutter Doctor output </summary>

```
[✓] Flutter (Channel stable, 3.19.1, on Microsoft Windows [Version 10.0.22631.3374], locale
    en-GB)
    • Flutter version 3.19.1 on channel stable at C:\Users\miche\fvm\versions\3.19.1
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision abb292a07e (7 weeks ago), 2024-02-20 14:35:05 -0800
    • Engine revision 04817c99c9
    • Dart version 3.3.0
    • DevTools version 2.31.1

[✓] Windows Version (Installed version of Windows is version 10 or higher)

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at C:\development\Android\Sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = C:\development\Android\Sdk
    • Java binary at: C:\Users\miche\AppData\Local\Programs\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.9+0--11185874)
    • All Android licenses accepted.

[✗] Chrome - develop for the web (Cannot find Chrome executable at
    .\Google\Chrome\Application\chrome.exe)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[!] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.9.4)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.9.34714.143
    • Windows 10 SDK version 10.0.22621.0
    ✗ Visual Studio is missing necessary components. Please re-run the Visual Studio installer
      for the "Desktop development with C++" workload, and include these components:
        MSVC v142 - VS 2019 C++ x64/x86 build tools
         - If there are multiple build tool versions available, install the latest
        C++ CMake tools for Windows
        Windows 10 SDK

[✓] Android Studio (version 2023.2)
    • Android Studio at C:\Users\miche\AppData\Local\Programs\Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.9+0--11185874)

[✓] Connected device (3 available)
    • SM G991B (mobile) • R3CR403EPAV • android-arm64  • Android 14 (API 34)
    • Windows (desktop) • windows     • windows-x64    • Microsoft Windows [Version
      10.0.22631.3374]
    • Edge (web)        • edge        • web-javascript • Microsoft Edge 123.0.2420.81

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.
```

</details>

## Additional notes

- I am using FVM; I am not sure it's relevant to the problem, I don't think it is but I wanted to mention it regardless.
- Potentially related? #6196

**Comment by helin24** [comment]
I'm not familiar with this concept of a dev drive. Is this windows-specific? A place to store source code, generally? Do you use this instead of the C drive for saving space?

**Comment by mvarendorff2** [comment]
@helin24 yeah, it's windows-specific. The Dev drive is a feature made available by Microsoft Dev Home and essentially consists of a partition with a more linux-like filesystem with reduced functionality but improved performance for lots of file reads. There is some additional information [here on SuperUser](https://superuser.com/a/1825686) and [here on Microsoft's devblog](https://devblogs.microsoft.com/visualstudio/devdrive/).

Before you dig into this, let me double check that this is still relevant though because it's been 1.5 years, I didn't even remember opening this issue (although I do recall the troubles now) and I don't think I have had any issues with this recently.

**Comment by mvarendorff2** [comment]
Hey, I just double checked and the issue still persists. If I can provide any information or test potential fixes, etc. do let me know!

**Comment by helin24** [comment]
Okay, as a first step, I'd probably need to add some logs to our path translation logic and see what's going wrong on windows, or set up a windows machine to try to reproduce. I'll let you know how to test if I get a chance to do that soon.

**Comment by helin24** [comment]
@mvarendorff2 I added some logging into the plugin that should help us understand what's going on with your setup - could you follow the steps below and report back?

1. Install the latest dev channel plugin following [instructions](https://github.com/flutter/flutter-intellij/blob/main/docs/Dev-Channel.md)
2. Go to your flutter plugin settings and make sure "Allow logging of full file paths" is checked: <img width="840" height="717" alt="Image" src="https://github.com/user-attachments/assets/caf59f9e-b500-448d-942e-42ec3c19cfb7" />
3. Reproduce the debugging problem and send me the contents of your dash.log file (Help > Show log in finder). Send to helinx@google.com if you'd rather not post publicly, as this will show file paths from your machine.

**Comment by helin24** [comment]
This is something we need more info about before we can make progress. I'm going to move this to backlog until then.

**Comment by mvarendorff** [comment]
Hey there! Sorry for the radio silence - the other account was abandoned because I left that job and I just saw the notification by chance the other day when closing that account down for good.

I can still reproduce the issue however, I sent the log to the address you provided @helin24, let me know if you need anything else (either here or as a response to the mail).
